src/components/Employee.tsx

```diff
@@ -0,0 +1,69 @@
+import { Link, useLocation } from "react-router-dom";
+
+export interface EmployeeInterface {
+  id: string;
+  name: string;
+  picture: string;
+  jobTitle: string;
+  location: string;
+}
+
+type EmployeeThumbnailProps = {
+  employee: EmployeeInterface;
+};
+
+export const EmployeeThumbnail = ({
+  employee,
+}: EmployeeThumbnailProps): JSX.Element => (
+  <Link to={`/${employee.id}`} state={{ employee }}>
+    <div className="employee-card">
+      <img
+        className="employee-thumbnail"
+        src={employee.picture}
+        alt={employee.name}
+      />
+      <div className="employee-info">
+        <h3 className="employee-bold--thumbnail">{employee.name}</h3>
+        <p className="employee-light--thumbnail">{employee.jobTitle}</p>
+      </div>
+    </div>
+  </Link>
+);
+
+const Employee = (): JSX.Element => {
+  const location = useLocation();
+  const { employee }: any = location.state;
```

Comment:
Why the type is set to `any` here? Shouldn't it be `EmployeeInterface`?

Reply:
If I set it to `EmployeeInterface`, I get the errors `Type 'unknown' is not assignable to type 'EmployeeInterface'` and `Property 'employee' does not exist on type 'EmployeeInterface'`.
If I don't set it anything, I get the error `Property 'employee' does not exist on type 'unknown'`.

I tried [this solution](https://thewebdev.info/2022/03/09/how-to-add-location-state-to-a-component-with-react-router-and-typescript/), but I am unable to set the type in `useLocation()` and it says `Expected 0 type arguments, but got 1`.

I can type cast it as such:
```js
const { employee } = location.state as { employee: EmployeeInterface };
```

Reply:
Yeah, it looks like casting it is the only solution (https://github.com/remix-run/react-router/issues/8503).